llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp

```diff
@@ -449,6 +449,15 @@ Register SPIRVGlobalRegistry::getOrCreateCompositeOrNull(
   return Res;
 }
 
+Register SPIRVGlobalRegistry::getOrCreateConstScalarOrVector(
+    uint64_t Val, MachineInstr &I, SPIRVType *SpvType,
+    const SPIRVInstrInfo &TII, bool ZeroAsNull) {
+  if (SpvType->getOpcode() == SPIRV::OpTypeVector)
+    return getOrCreateConstVector(Val, I, SpvType, TII, ZeroAsNull);
+  else
+    return getOrCreateConstInt(Val, I, SpvType, TII, ZeroAsNull);
```

Comment:
Should we be checking `SpvType` if it is float or int like you did for vector before deciding to create a const int?

Reply:
No, because the Val is a uint64_t so its assumed to be an int. Following how 'getOrCreateConstVectr' is written, someone would write an overload of this that takes a float if they wanted it for floats.